### DISCUSSION TITLE
Filament logo 3D Model

### DISCUSSION THREAD

**Discussion by ThomasGorisse** [google/filament#4010] [Ideas]
Hi,

Just a little quick think:
Have you made any 3D version of the Filament logo ? If not, it would be great to have it for multiple purposes including samples with the bulb glass material.
And I have to say that I would like to use it for a featured screenshot picture of a sample AR app using my SDK and showing the 3D rendered Filament logo appearing partially behind a real world 3D printed ARCore Logo (Depth API applied to Filament rendering)
BTW, is there any worries about the usage of the Filament logo on my repo ?[https://github.com/ThomasGorisse/realitycore/blob/main/docs/images/logos/realitycore_logo.png](https://github.com/ThomasGorisse/realitycore/blob/main/docs/images/logos/realitycore_logo.png)

**Comment by ThomasGorisse** [comment]
Rainy Saturday morning in France. I made the ARCore one. Will do the Filament model during the 3D print if it's raining again ;-)

[logo_arcore_glb.zip](https://github.com/google/filament/files/6526295/logo_arcore_glb.zip)

![image](https://user-images.githubusercontent.com/6597529/119224136-51fd9e80-bafd-11eb-8d6a-f9b2b2c6fb7b.png)

**Reply by romainguy** [reply]
We haven't made a 3D version of the Filament logo. You can use our 2D logo if you please but I don't know whether you can do the same with the ARCore logo.

**Reply by ThomasGorisse** [reply]
Work in progress

![image](https://user-images.githubusercontent.com/6597529/119260376-d0784000-bbd2-11eb-857f-d90d1331df53.png)
I love Blender

**Reply by ThomasGorisse** [reply]
Do you know the font name used for the "Filament" text ?
Don't know yet how I will include it in the model to keep the original text logo philosophy. Maybe like the trade mark on a real bulb = printed on the glass.

**Reply by romainguy** [reply]
The font is "SignPainter".

**Comment by ThomasGorisse** [comment]
Here we are!

![image](https://user-images.githubusercontent.com/6597529/120069130-92709580-c084-11eb-82ff-9af48deed243.png)
![image](https://user-images.githubusercontent.com/6597529/120069207-f5fac300-c084-11eb-8b8c-a8b5cca8c727.png)
![image](https://user-images.githubusercontent.com/6597529/120069700-8df9ac00-c087-11eb-87b4-4bbfc3b1bdc4.png)

Here are the .glb and .blend : [Filament.zip](https://github.com/google/filament/files/6564628/Filament.zip)

Feel free to use/modify/... it. I just made it for fun.

The glass material isn't representative at all. There may be some modifications to do on it for better results.

To export the model, since blender gltf export include not visible/not rendered checked objects, please select visible ones and check `Selected Objects Only` on the export dialog.

**Reply by romainguy** [reply]
Thank you! The glass could be made to look a lot better by using the KHR_materials_transmission extension. Otherwise any alpha value is also applied to specular lighting.

**Reply by ThomasGorisse** [reply]
![image](https://user-images.githubusercontent.com/6597529/120084538-5e6f9180-c0d1-11eb-835c-735974244146.png)
![image](https://user-images.githubusercontent.com/6597529/120084874-156d0c80-c0d4-11eb-9f49-6ce16051ff8f.png)
![image](https://user-images.githubusercontent.com/6597529/120084897-52d19a00-c0d4-11eb-8f2f-92e1a54ae600.png)
![image](https://user-images.githubusercontent.com/6597529/120085172-1e5edd80-c0d6-11eb-81e9-6ec0ba7157b9.png)

Better?

**Reply by ThomasGorisse** [reply]
I will make the test with HDR Lighting/IBL
I just don't know yet why the orange light near the filament inside the bulb is not rendered with Sceneform.

**Reply by romainguy** [reply]
If you want we could add this model to our source tree if you open a PR. `assets/models` would be a great place for it.

**Reply by ThomasGorisse** [reply]
Sure. I'll PR it.
Just have some rendering issues **on a Pixel 4a** with the 1.19.25 and this glb :
[logo_filament-v1.0.0.glb.zip](https://github.com/google/filament/files/6577753/logo_filament-v1.0.0.glb.zip)
I'll try to figure out where does the problem comes from before pushing.
Should I also commit the .blend?

**Reply by romainguy** [reply]
What issues do you have in .25? I'm seeing transparency issues with master and our desktop viewer. Are you using double-sided geometry for the bulb or something else?

For the .blend it's up to you, but we'll gladly accept it.

I'm surprised by how big the .glb is. 17 MiB without textures seems excessive. From a quick look in Blender, it seems the text part of the logo is using a LOT of triangles. It might be worth reducing the triangle count there.

**Reply by ThomasGorisse** [reply]
> What issues do you have in .25?

I have some full SurfaceView white artifcats maybe due to Sceneform if you can't see it with the modelviewer.

> I'm seeing transparency issues with master and our desktop viewer. Are you using double-sided geometry for the bulb or something else?

Fixed. I forgot to remove Backface Culling used for tests.

> For the .blend it's up to you, but we'll gladly accept it.

I'll push a clean version without applying modifiers to possibly modify some parts later.

> I'm surprised by how big the .glb is. 17 MiB without textures seems excessive. From a quick look in Blender, it seems the text part of the logo is using a LOT of triangles. It might be worth reducing the triangle count there.

You are right, I have 350k faces coming from the filament object which is a circle with a screw modifier applied to it.
I don't know if I can reduce the vertices count of the original circle without having to recreate it from scratch with the same modifiers.
And 50K for the text which is not converted to a mesh yet.

**Reply by ThomasGorisse** [reply]
It shouldn't be a problem to reduce the face count of a bulb since I already managed to render monuments with Sceneform coming from topographics drones with an original size of more than 1Go reduced to 200Mo (thanks to blender decimate and other tricks) without low polying it to much.
By the way, beyond the professional usage, rendering a complete monument and being able to visit it and view normally invisible little details while walking inside it on an AR experience is amazing. (Users just need more than a living room space to use it)

**Comment by ThomasGorisse** [comment]
Here is the result : [logo_filament-v1.0.1.glb.zip](https://github.com/google/filament/files/6580253/logo_filament-v1.0.1.glb.zip)

![GIF 01-06-2021 21-49-19](https://user-images.githubusercontent.com/6597529/120382081-86563380-c323-11eb-866e-e7e553c7c90a.gif)
![image](https://user-images.githubusercontent.com/6597529/120383688-7a6b7100-c325-11eb-955e-97664c694d00.png)
Sorry for using another rendering engine for the snapshot but can't find a good link for a web drag and drop version.

Let me know if everything is OK for you and I'll make the PR

**Reply by romainguy** [reply]
The online glTF viewer is here: https://google.github.io/filament/viewer/

**Reply by romainguy** [reply]
BTW there are weird triangulation bugs in the text, at some angles you can see the reflections are broken. It looks like normals that are not correctly computed across neighboring triangles?

**Reply by ThomasGorisse** [reply]
You have good eyes.
I tried some bad mesh cleanup and decimated the text too much.
To be sincere, I was hoping nobody will notice it since I couldn't undo it. :-)

We (developers) must be grateful to version control for the ability to rollback changes on new versions, it's must be a nightmare for 3D designers.
